code/game/objects/items/gift.dm

```diff
@@ -18,25 +18,35 @@ GLOBAL_LIST_EMPTY(possible_gifts)
 	item_state = "gift"
 	resistance_flags = FLAMMABLE
 
-/obj/item/a_gift/New()
-	..()
+	var/obj/item/contains_type
+
+/obj/item/a_gift/Initialize()
+	. = ..()
 	pixel_x = rand(-10,10)
 	pixel_y = rand(-10,10)
 	icon_state = "giftdeliverypackage[rand(1,5)]"
 
+	contains_type = get_gift_type()
+
 /obj/item/a_gift/suicide_act(mob/user)
 	user.visible_message("<span class='suicide'>[user] peeks inside [src] and cries [user.p_them()]self to death! It looks like [user.p_they()] [user.p_were()] on the naughty list...</span>")
 	return (BRUTELOSS)
 
+/obj/item/a_gift/examine(mob/M)
+	. = ..()
+	if(M.has_trait(TRAIT_PRESENT_VISION) || isobserver(M))
+		to_chat(M, "<span class='notice'>It contains \a [initial(contains_type.name)].</span>")
+
 /obj/item/a_gift/attack_self(mob/M)
-	if(M && M.mind && M.mind.special_role == "Santa")
+	if(M.has_trait(TRAIT_CANNOT_OPEN_PRESENTS))
 		to_chat(M, "<span class='warning'>You're supposed to be spreading gifts, not opening them yourself!</span>")
 		return
 
-	var/gift_type = get_gift_type()
-
 	qdel(src)
-	var/obj/item/I = new gift_type(M)
+
+	var/obj/item/I = new contains_type(get_turf(M))
+	M.visible_message("<span class='notice'>[M] unwraps \the [src], finding \a [I] inside!</span>")
+	I.investigate_log("([I.type]) was found in a present by [key_name(M)].", INVESTIGATE_PRESENTS)
```

Comment:
Most of the presents are pretty benign aren't they? just log it under the mob's attack logs or something. It certainly doesn't need its own panel.

I'm not sure what's to be gained by this information that wouldn't already be covered by fingerprints on the item and a mob's attack logs though.

Reply:
Because sometimes people will get the item that summons a singularity, and we'll need to know who it was.

When #41924 gets merged, it won't even be taking up a slot in the active investigate options unless someone's been gift opening anyway.

Reply:
Since the experimentor does this i'll be okay with it.